controllers/gateway/api_controller_integration_test.go

```diff
@@ -1395,6 +1395,330 @@ var _ = Describe("APIRule Controller", Serial, func() {
+		}
+
+		It("should not create an APIRule with an empty gateway", func() {
+			invalidHelper("")
+		})
+
+		It("should not create an APIRule with too long gateway namespace name", func() {
+			invalidHelper("insane-very-long-namespace-name-exceeding-sixty-three-characters/validname")
+		})
+
+		It("should not create an APIRule with too long gateway name", func() {
+			invalidHelper("validnamespace/insane-very-long-namespace-name-exceeding-sixty-three-characters")
+		})
+
+		It("should not create an APIRule with just the namespace", func() {
+			invalidHelper("validnamespace/")
+		})
+
+		It("should not create an APIRule with just the gateway name", func() {
+			invalidHelper("/validgateway")
+		})
+
+		It("should not create an APIRule with double slashed gateway name", func() {
+			invalidHelper("namespace//gateway")
+		})
+	})
+
+	Context("when creating APIRule in version v2alpha1 hosts should be FQDN", Ordered, func() {
+		It("should not create an APIRule with an empty host", func() {
+			// given
+			apiRuleName := generateTestName(testNameBase, testIDLength)
+			serviceName := testServiceNameBase
+			serviceHost := gatewayv2alpha1.Host("")
+			serviceHosts := []*gatewayv2alpha1.Host{&serviceHost}
+
+			rule := testRulev2alpha1("/img", []gatewayv2alpha1.HttpMethod{http.MethodGet})
+			rule.NoAuth = ptr.To(true)
+			apiRule := testApiRulev2alpha1(apiRuleName, testNamespace, serviceName, testNamespace, serviceHosts, testServicePort, []gatewayv2alpha1.Rule{rule})
+			svc := testService(serviceName, testNamespace, testServicePort)
+
+			// when
+			Expect(c.Create(context.Background(), svc)).Should(Succeed())
+			err := c.Create(context.Background(), apiRule)
+			defer func() {
+				apiRulev2alpha1Teardown(apiRule)
+				serviceTeardown(svc)
+			}()
+			Expect(err).Should(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("spec.hosts[0]: Invalid value: \"string\": Host is not Fully Qualified Domain Name"))
+		})
+
+		It("should create an APIRule with a FQDN host", func() {
+			// given
+			apiRuleName := generateTestName(testNameBase, testIDLength)
+			serviceName := testServiceNameBase
+			serviceHost := gatewayv2alpha1.Host("example.com")
+			serviceHosts := []*gatewayv2alpha1.Host{&serviceHost}
+
+			rule := testRulev2alpha1("/img", []gatewayv2alpha1.HttpMethod{http.MethodGet})
+			rule.NoAuth = ptr.To(true)
+			apiRule := testApiRulev2alpha1(apiRuleName, testNamespace, serviceName, testNamespace, serviceHosts, testServicePort, []gatewayv2alpha1.Rule{rule})
+			svc := testService(serviceName, testNamespace, testServicePort)
+
+			// when
+			Expect(c.Create(context.Background(), svc)).Should(Succeed())
+			Expect(c.Create(context.Background(), apiRule)).Should(Succeed())
+			defer func() {
+				apiRulev2alpha1Teardown(apiRule)
+				serviceTeardown(svc)
+			}()
+		})
+
+		It("should not create an APIRule with host name without domain", func() {
```

Comment:
There's a lot of stuff duplicated in those tests, maybe we could make them into a table, or at least create some setup func that would take just the host?